Compile error with GraphViz 2.36.0

I get an error when I try to compile GammaRay (same with 2.0.0, 2.0 and master).
This seems to be related with GraphViz. I have version 2.36.0 installed.
With GraphViz 2.34.0, it works.

```
[ 79%] Building CXX object plugins/statemachineviewer/CMakeFiles/gammaray_statemachineviewer_ui_plugin.dir/gvgraph/gvutils.cpp.o
In file included from /usr/include/graphviz/cgraph.h:17:0,
                 from /usr/include/graphviz/types.h:40,
                 from /home/cbuehler/gammaray/src/GammaRay/plugins/statemachineviewer/gvgraph/gvutils.h:25,
                 from /home/cbuehler/gammaray/src/GammaRay/plugins/statemachineviewer/gvgraph/gvutils.cpp:24:
/usr/include/graphviz/cdt.h:27:12: error: previous declaration of 'int memcmp(const void*, const void*, size_t)' with 'C++' linkage
 extern int  memcmp _ARG_((const Void_t*, const Void_t*, size_t));
            ^
In file included from /usr/include/qt/QtCore/qarraydata.h:46:0,
                 from /usr/include/qt/QtCore/qbytearray.h:47,
                 from /usr/include/qt/QtCore/qstring.h:46,
                 from /usr/include/qt/QtCore/QString:1,
                 from /home/cbuehler/gammaray/src/GammaRay/plugins/statemachineviewer/gvgraph/gvutils.h:27,
                 from /home/cbuehler/gammaray/src/GammaRay/plugins/statemachineviewer/gvgraph/gvutils.cpp:24:
/usr/include/string.h:70:77: error: conflicts with new declaration with 'C' linkage
      __THROW __attribute_pure__ __nonnull ((1, 2));
                                                                             ^
In file included from /usr/include/graphviz/cgraph.h:17:0,
                 from /usr/include/graphviz/types.h:40,
                 from /home/cbuehler/gammaray/src/GammaRay/plugins/statemachineviewer/gvgraph/gvutils.h:25,
                 from /home/cbuehler/gammaray/src/GammaRay/plugins/statemachineviewer/gvgraph/gvutils.cpp:24:
/usr/include/graphviz/cdt.h:28:12: error: previous declaration of 'int strcmp(const char*, const char*)' with 'C++' linkage
 extern int  strcmp _ARG_((const char*, const char*));
            ^
In file included from /usr/include/qt/QtCore/qarraydata.h:46:0,
                 from /usr/include/qt/QtCore/qbytearray.h:47,
                 from /usr/include/qt/QtCore/qstring.h:46,
                 from /usr/include/qt/QtCore/QString:1,
                 from /home/cbuehler/gammaray/src/GammaRay/plugins/statemachineviewer/gvgraph/gvutils.h:27,
                 from /home/cbuehler/gammaray/src/GammaRay/plugins/statemachineviewer/gvgraph/gvutils.cpp:24:
/usr/include/string.h:145:77: error: conflicts with new declaration with 'C' linkage
      __THROW __attribute_pure__ __nonnull ((1, 2));
                                                                             ^
plugins/statemachineviewer/CMakeFiles/gammaray_statemachineviewer_ui_plugin.dir/build.make:200: recipe for target 'plugins/statemachineviewer/CMakeFiles/gammaray_statemachineviewer_ui_plugin.dir/gvgraph/gvutils.cpp.o' failed
make[2]: *** [plugins/statemachineviewer/CMakeFiles/gammaray_statemachineviewer_ui_plugin.dir/gvgraph/gvutils.cpp.o] Error 1
CMakeFiles/Makefile2:2798: recipe for target 'plugins/statemachineviewer/CMakeFiles/gammaray_statemachineviewer_ui_plugin.dir/all' failed
make[1]: *** [plugins/statemachineviewer/CMakeFiles/gammaray_statemachineviewer_ui_plugin.dir/all] Error 2
Makefile:126: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/cmake/FindGraphviz.cmake
+++ b/cmake/FindGraphviz.cmake
@@ -8,6 +8,7 @@
 #  GRAPHVIZ_MAJOR_VERSION = The library major version number
 #  GRAPHVIZ_MINOR_VERSION = The library minor version number
 #  GRAPHVIZ_PATCH_VERSION = The library patch version number
+#  GRAPHVIZ_COMPILE_FLAGS = List of compile flags needed by the GraphViz installation headers
 #
 # This module reads hints about search locations from the following env variables:
 #  GRAPHVIZ_ROOT          - Graphviz installation prefix
@@ -145,6 +146,12 @@ if(GRAPHVIZ_FOUND)
     list(GET VL 0 GRAPHVIZ_MAJOR_VERSION)
     list(GET VL 1 GRAPHVIZ_MINOR_VERSION)
     list(GET VL 2 GRAPHVIZ_PATCH_VERSION)
+  endif()
+
+  set(GRAPHVIZ_COMPILE_FLAGS "")
+  check_include_files(string.h HAVE_STRING_H)
+  if (HAVE_STRING_H)
+    list(APPEND GRAPHVIZ_COMPILE_FLAGS "-DHAVE_STRING_H=1")
   endif()
 
   if(NOT GRAPHVIZ_FIND_QUIETLY)

--- a/plugins/statemachineviewer/CMakeLists.txt
+++ b/plugins/statemachineviewer/CMakeLists.txt
@@ -69,6 +69,7 @@ else()
     ${GRAPHVIZ_GRAPH_LIBRARY}
   )
 endif()
+add_definitions(${GRAPHVIZ_COMPILE_FLAGS})
 
 qt4_wrap_ui(gammaray_statemachineviewer_ui_plugin_srcs statemachineviewer.ui)
 

